apiserver/facades/client/application/backend.go

```diff
@@ -4,39 +4,13 @@
 package application
 
 import (
-	"github.com/juju/schema"
-
 	"github.com/juju/juju/apiserver/common/storagecommon"
-	coreconfig "github.com/juju/juju/core/config"
-	"github.com/juju/juju/core/constraints"
 	coremodel "github.com/juju/juju/core/model"
 	"github.com/juju/juju/core/network"
-	"github.com/juju/juju/core/objectstore"
 	"github.com/juju/juju/internal/charm"
-	"github.com/juju/juju/internal/configschema"
 	"github.com/juju/juju/state"
 )
 
-// Backend defines the state functionality required by the application
-// facade. For details on the methods, see the methods on state.State
-// with the same names.
-type Backend interface {
-	Application(string) (Application, error)
-	ApplyOperation(state.ModelOperation) error
-	AddApplication(state.AddApplicationArgs, objectstore.ObjectStore) (Application, error)
-}
-
-// Application defines a subset of the functionality provided by the
-// state.Application type, as required by the application facade. For
-// details on the methods, see the methods on state.Application with
-// the same names.
-type Application interface {
-	SetCharm(state.SetCharmConfig, objectstore.ObjectStore) error
-	SetConstraints(constraints.Value) error
-	UpdateCharmConfig(charm.Settings) error
-	UpdateApplicationConfig(coreconfig.ConfigAttributes, []string, configschema.Fields, schema.Defaults) error
-}
```

Comment:
Can any of these go from `state`?

Reply:
I'm leaving all pure state removal for next patch (what I do is to first fix the impacted facades/workers and then remove).